lib/fitfree.gi

```diff
@@ -1398,6 +1398,31 @@ local l;
   fi;
 end);
 
+
+#############################################################################
+##
+#M  SylowSubgroupOp( <G>, <pi> )
+##
+## Fitting free approach
+##
+InstallMethod( SylowSubgroupOp, "fitting free",true,
+    [ IsGroup and CanComputeFittingFree,IsPosInt ],
+    -1, # rank lower than anything that would imply CanComputeFittingFree
+function(G,pi)
+local l;
+  if Set(Factors(Size(G)))=[pi] then 
+    SetIsPGroup(G,true);
```

Comment:
Ideally the prime should also be set (some code gets confused if it isn't, or at least it used to be that way)
```suggestion
    SetIsPGroup(G,true);
    SetPrimePGroup(G, pi);
```